### PR TITLE
Clock Abstraction Layer

### PR DESCRIPTION
**Pull request by rlubos**

### CODE REVIEW

**Comment by LuDuda** [comment]
Great work and improvement for the driver!

One topic to discuss in SoftDevice RAAL:

Right now we act on SoftDevice `NRF_EVT_HFLCKSTARTED` inside the nrf_raal_softdevice.c.
I think to make it more consistent with other RAALs i would remove handling of `NRF_EVT_HFCLKSTARTED` from the `nrf_raal_softdevice_soc_evt_handler` and define `nrf_drv_radio802154_clock_hfclk_ready` function instead.

What do you think?

**Comment by rlubos** [comment]
Sounds like a good improvement. I'll update the PR.

**Comment by hubertmis** [comment]
I like this improvement. Did you test it with all supported RAAL implementations?

**Comment by rlubos** [comment]
Standalone only. I'll check the softdevice version with SDK.

**Comment by hubertmis** [comment]
What do you mean by 'standalone'? Only single-phy? Or single-phy + simulator?

**Comment by rlubos** [comment]
single-phy

File: src/raal/softdevice/nrf_raal_softdevice.c [hunk 1]
```diff
@@ -722,3 +714,12 @@ void nrf_raal_critical_section_exit(void)
 
     nrf_drv_radio802154_log(EVENT_TRACE_EXIT, FUNCTION_RAAL_CRIT_SECT_EXIT);
 }
+
```
Comment (LuDuda) [inline src/raal/softdevice/nrf_raal_softdevice.c]: extra line? Is it intentional?

**Comment by rlubos** [comment]
Looks like it's stable now - I think it's ready to merge.

**Comment by hubertmis** [comment]
Please, rebase this branch to master

**Pull request merged**